Framework/PythonInterface/mantid/plots/datafunctions.py

```diff
@@ -342,6 +342,41 @@ def get_spectrum(workspace, wkspIndex, normalize_by_bin_width, withDy=False, wit
     return x, y, dy, dx
 
 
+def get_detectors_indices(workspace):
+    """
+    Find the detectors' spectrum indices
+    (ie every detector which is not a monitor)
+
+    :param workspace: a Workspace2D or an EventWorkspace
+    :return : the detectors' spectrum indices as a range if possible, else as a numpy array
+    """
+    total_range = workspace.getNumberHistograms()
+    spectrum_info = workspace.spectrumInfo()
+    monitors_indices = [i for i in range(total_range) if spectrum_info.hasDetectors(i) and spectrum_info.isMonitor(i)]
+    monitor_count = len(monitors_indices)
+
+    # If possible, ie the detectors' indices are continuous, we return a range.
+    # If not, we return a numpy array
+    range_start = -1
+    range_end = total_range
+    is_range = True
+    for index, i in enumerate(monitors_indices):
```

Comment:
Can we avoid one character variable names, e.g. i here which can't be index as there is another index variable, so it needs a better name.  is it monitor_index?